### DISCUSSION TITLE
Setting drag coefficients for rigid bodies with Langevin dynamics

### DISCUSSION THREAD

**Discussion by crisbutu** [glotzerlab/hoomd-blue#1768] [Support] [answered]
Hello, 

I have a question related to handling drag for rigid bodies when using Langevin dynamics. 

1. Is the drag coefficient of a rigid body determined solely by the gamma parameter of its center bead, ignoring constituent beads? 

2. I'm uncertain about the interpretation of the gamma_r parameter (rotational drag coefficient). The documentation for version 2.9.7, which I'm using, specifies its units as mass/time (the same for v3). With the default setting being gamma_r = (1,1,1), does this imply that the rotational diffusivity of the rigid body is equivalent to that of a sphere, regardless of its actual shape? Or does the internal handling consider the geometry of the rigid body to produce an expected drag tensor with units of mass*length^2/time?

Specifically, I want to implement rigid rods and have set the moment of inertia for the central bead as that of a cylinder. I want to ensure that the rotational diffusivity about the axial direction differs from the perpendicular directions, as expected for a rod. Could anyone advise on how to correctly configure this?

Any help would be greatly appreciated. Thank you!

**Comment by joaander** [comment, marked as answer]
The three components of `gamma_r` are the rotational drag coefficients about the three axes of rotation (x, y, z). `gamma_r` has units of 1/time because each component appears in the equation of motion as: 
$$I \frac{d\vec{L}}{dt} = \vec{\tau}_C  - \gamma_r \cdot \vec{L}  + \vec{\tau}_R$$
(this equation is in the [documenation](https://hoomd-blue.readthedocs.io/en/v4.6.0/module-md-methods.html#hoomd.md.methods.Langevin)). Note that `\cdot` here is an element-wise multiplication - not a dot product.

The angular momentum $L$ has units [mass][length]^2[time]^(-1) and torque $\tau$ has units [mass][length]^2[time]^(-2), so `gamma_r` must have units of [time]^(-1).

However, looking at the code - the documentation for this may be incorrect: https://github.com/glotzerlab/hoomd-blue/blob/103d1b4ce7b94891e54981b4dd8ee936360059a8/hoomd/md/TwoStepLangevin.cc#L316-L355
I don't have time right now to fix it. Look at `rand_x` (which is a function gamma_r.x) and `gamma_r.x * (s.x / I.x)` and figure out what units it should be. Feel free to file a pull request that makes the correction to the docs.

**Reply by zmcdargh** [reply]
I think this equation is a mistake, and the `L` here is supposed to be angular velocity, not angular momentum (the net torque on a body is just `dL/dT`, not `I*dL/dt`). Making this change resolves the issue with the dimensions of `gamma_r`, the discrepancy between the code and the docs, and a few inconsistencies in the docs. Either I or @crisbutu will do a PR to fix this. Thanks for your help!

**Reply by joaander** [reply]
Thanks! I look forward to the PR.
